### PR TITLE
Buffs light replacer and makes it easier to use

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscadd: The Janitor's Union has mandated additional features to station
light replacer devices.
rscadd: The light replacer now can be used on the floor to replace
the bulbs of any lights on that tile.
rscadd: The light replacer now eats replaced bulbs, and after a certain
number of bulb fragments (four) will recycle them into a new bulb.
tweak: The description of the light replacer is now more informative.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
:+1:

File: code/game/objects/items/devices/lightreplacer.dm [hunk 1]
```diff
@@ -220,7 +241,25 @@
...
+	for(var/atom/A in T)
+		if(!CanUse(U))
+			break
+		used = TRUE
+		if(istype(A, /obj/machinery/light))
+			ReplaceLight(A, U)
+
+	if(!used)
+		U << failmsg
+		return
```
Comment (RemieRichards) [inline code/game/objects/items/devices/lightreplacer.dm]: trailing return, remove it please.

**Comment by RandomMarine** [comment]
>eating light bulbs with replacer
>when autolathe recycling is 1:1

Worse than even dumping them into the disposals recycler.

**Comment by coiax** [comment]
Yep, but it doesn't leave a mess. And barely anyone picks up the broken bulbs, so that's a 0% efficency. At least we're getting something out of it now.

**Comment by PKPenguin321** [comment]
i miss the pink thumbs tbh

**Comment by CPTANT** [comment]
Just give it infinite charge, nobody cares about the negligible material cost of replacing lights.

**Comment by coiax** [comment]
Carry a stack of glass with you, or use a box of lightbulbs on the replacer for it to suck them all in.

**Comment by octareenroon91** [comment]
Maybe give the replacer a cell for making bulbs from raw materials?

**Comment by coiax** [comment]
You can already insert glass sheets into the replacer, to the tune of 5 bulbs per sheet.

**Comment by RandomMarine** [comment]
For consistency and realism's sake, broken lights should have reduced MAT_GLASS count. This way there's no more or less efficient method of recycling broken lights.

**Comment by coiax** [comment]
Good idea, but that's for another PR, probably entitled "Super exciting broken lightbulb material nerfs"

**Pull request merged**